From 194d50e9f857dc6fe1fa7db4522c162f2e40b0b8 Mon Sep 17 00:00:00 2001
From: Weijie Gao <hackpascal@gmail.com>
Date: Mon, 27 Feb 2017 17:08:06 +0800
Subject: [PATCH 2/5] MIPS: add realtek mach support

This patch adds mach support for Realtek's RLX-based SoCs.

Signed-off-by: Weijie Gao <hackpascal@gmail.com>
---
 arch/mips/Kbuild.platforms |  1 +
 arch/mips/Kconfig          | 22 ++++++++++++++++++++++
 arch/mips/pci/Makefile     |  1 +
 3 files changed, 24 insertions(+)

Index: linux-4.13/arch/mips/Kbuild.platforms
===================================================================
--- linux-4.13.orig/arch/mips/Kbuild.platforms	2018-01-14 14:53:16.419823751 +0800
+++ linux-4.13/arch/mips/Kbuild.platforms	2018-01-14 14:53:16.415823724 +0800
@@ -26,6 +26,7 @@
 platforms += pmcs-msp71xx
 platforms += pnx833x
 platforms += ralink
+platforms += realtek
 platforms += rb532
 platforms += sgi-ip22
 platforms += sgi-ip27
Index: linux-4.13/arch/mips/Kconfig
===================================================================
--- linux-4.13.orig/arch/mips/Kconfig	2018-01-14 14:53:16.419823751 +0800
+++ linux-4.13/arch/mips/Kconfig	2018-01-14 14:53:16.415823724 +0800
@@ -1007,6 +1007,27 @@
 	help
 	  This option supports guest running under ????
 
+config REALTEK
+	bool "Realtek RLX based SoC support"
+	select BOOT_RAW
+	select DMA_NONCOHERENT
+	select IRQ_MIPS_CPU
+	select IRQ_DOMAIN
+	select HW_HAS_PCI
+	select MIPS_MACHINE
+	select HAVE_CLK
+	select COMMON_CLK
+	select CLKDEV_LOOKUP
+	select GENERIC_CSUM
+	select ARCH_REQUIRE_GPIOLIB
+	select SYS_HAS_CPU_R3000
+	select SYS_SUPPORTS_BIG_ENDIAN
+	select SYS_SUPPORTS_MIPS16
+	select SYS_SUPPORTS_32BIT_KERNEL
+	select SYS_HAS_EARLY_PRINTK
+	help
+	  Support for Realtek RLX SoC based boards
+
 endchoice
 
 source "arch/mips/alchemy/Kconfig"
@@ -1034,6 +1055,7 @@
 source "arch/mips/netlogic/Kconfig"
 source "arch/mips/paravirt/Kconfig"
 source "arch/mips/xilfpga/Kconfig"
+source "arch/mips/realtek/Kconfig"
 
 endmenu
 
Index: linux-4.13/arch/mips/pci/Makefile
===================================================================
--- linux-4.13.orig/arch/mips/pci/Makefile	2018-01-14 14:53:16.419823751 +0800
+++ linux-4.13/arch/mips/pci/Makefile	2018-01-14 14:53:16.419823751 +0800
@@ -25,6 +25,7 @@
 obj-$(CONFIG_SOC_AR71XX)	+= pci-ar71xx.o
 obj-$(CONFIG_PCI_AR724X)	+= pci-ar724x.o
 obj-$(CONFIG_MIPS_PCI_VIRTIO)	+= pci-virtio-guest.o
+obj-$(CONFIG_REALTEK)		+= pci-realtek.o
 #
 # These are still pretty much in the old state, watch, go blind.
 #
